src/home_assistant.rs

```diff
@@ -128,11 +143,30 @@ impl Config {
                 Register::AcChargeEndSocLimit,
                 "Charge From AC Upper Limit %",
             )?,
+            self.time_range("ac_charge/1", "AC Charge Timeslot 1")?,
+            self.time_range("ac_charge/2", "AC Charge Timeslot 2")?,
+            self.time_range("ac_charge/3", "AC Charge Timeslot 3")?,
+            self.time_range("charge_priority/1", "Charge Priority Timeslot 1")?,
+            self.time_range("charge_priority/2", "Charge Priority Timeslot 2")?,
+            self.time_range("charge_priority/3", "Charge Priority Timeslot 3")?,
+            self.time_range("forced_discharge/1", "Forced Discharge Timeslot 1")?,
+            self.time_range("forced_discharge/2", "Forced Discharge Timeslot 2")?,
+            self.time_range("forced_discharge/3", "Forced Discharge Timeslot 3")?,
         ];
 
         Ok(r)
     }
 
+    fn ha_discovery_topic(&self, kind: &str, name: &str) -> String {
+        format!(
+            "{}/{}/lxp_{}/{}/config",
+            self.mqtt_config.homeassistant().prefix(),
+            kind,
+            self.inverter.datalog(),
+            name.replace('/', "_"),
```

Comment:
Without this, the message for `ac_charge/1` ends up nested a level deeper than HA expects. It's not obvious so I decided to centralise it.

Reply:
Probably just deserves a code comment but yeah seems fine.

Reply:
Test is a very good idea :thumbsup: - will add before I take it off draft.

Reply:
I had a go at the test and struggled a bit :grimacing:. It can't go in in `tests/test_home_assistant.rs` because it's a private method, and I didn't like the idea of pulling that site's common code for setup into `src/home_assistant.rs`.

So, I bailed out and just added the comment, plus another of the HA discovery message tests that implicitly exercises this code :sweat_smile: